a3p-integration/proposals/z:acceptance/vaults.test.js

```diff
@@ -115,133 +95,327 @@ test.serial('remove IST', async t => {
-  const vaultCollateral = vaultData.locked.value;
-  t.log('vault collateral:', vaultCollateral);
+  const { collateral: collateralAfter } =
+    await getLastVaultFromAddress(USER1ADDR);
+  t.log('vault collateral after:', collateralAfter);
+
+  t.is(
+    collateralBefore,
+    collateralAfter - 1_000_000n,
+    'The vault Collateral should increase after adding some ATOM',
+  );
+});
+
+test.serial('add IST', async t => {
+  const { vaultID, debt: debtBefore } =
+    await getLastVaultFromAddress(USER1ADDR);
+  t.log('vault debt before:', debtBefore);
+
+  await adjustVault(USER1ADDR, vaultID, { giveMinted: 1.0 });
+
+  const { debt: debtAfter } = await getLastVaultFromAddress(USER1ADDR);
+  t.log('vault debt after:', debtAfter);
+
+  t.is(
+    debtAfter,
+    debtBefore - 1_000_000n,
+    'The vault Debt should decrease after adding some IST',
+  );
+});
+
+test.serial('close vault', async t => {
+  const { vaultID, collateral } = await getLastVaultFromAddress(USER1ADDR);
+  t.log('vault collateral:', collateral);
 
   const atomBalanceBefore = await getBalances([USER1ADDR], ATOM_DENOM);
   t.log('atom balance before', atomBalanceBefore);
 
-  await closeVault(USER1ADDR, vaultID, 6.03);
-  await waitForBlock();
+  await closeVault(USER1ADDR, vaultID, 6.035);
 
   const atomBalanceAfter = await getBalances([USER1ADDR], ATOM_DENOM);
   t.log('atom balance after', atomBalanceAfter);
 
-  vaultData = await getContractInfo(vaultPath, { agoric, prefix: '' });
-  const vaultState = vaultData.vaultState;
-  t.log('vault state:', vaultState);
+  const { state } = await getLastVaultFromAddress(USER1ADDR);
+  t.log('vault state:', state);
 
   t.is(
     atomBalanceAfter,
-    atomBalanceBefore + vaultCollateral,
+    atomBalanceBefore + collateral,
     'The ATOM balance should increase by the vault collateral amount',
   );
-  t.is(vaultState, 'closed', 'The vault should be in the "closed" state.');
+  t.is(state, 'closed', 'The vault should be in the "closed" state.');
 });
 
-test.serial('open second vault', async t => {
-  const user2Address = await getUser('user2');
-  await bankSend(user2Address, `20000000${ATOM_DENOM}`);
-
-  const activeVaultsBefore = await agopsVaults(user2Address);
+test.serial(
+  'user cannot open a vault under the minimum initial debt',
+  async t => {
+    await bankSend(GOV1ADDR, `200000000000000000${ATOM_DENOM}`);
+    const activeVaultsBefore = await agopsVaults(GOV1ADDR);
+    t.log('active vaults before:', activeVaultsBefore);
+
+    const minInitialDebt = await getMinInitialDebt();
+
+    const mint = minInitialDebt - 1n;
+    const collateral = mint * 2n;
+
+    await t.throwsAsync(
+      () => openVault(GOV1ADDR, mint.toString(), collateral.toString()),
+      {
+        message: new RegExp(
+          `Error: Vault creation requires a minInitialDebt of {"brand":"\\[Alleged: IST brand\\]","value":"\\[${minInitialDebt * 1_000_000n}n\\]"}`,
+        ),
```

Comment:
That seems like a lot of work to encode what's really irrelevant detail. If `openVault()` throws with a message that includes "Vault creation requires a minInitialDebt", that's a win. 

```suggestion
        message: /Error: Vault creation requires a minInitialDebt/,
```

We might want a unit test that verifies that the message is correct, but this is a system test; what we care about here is that it fails.  (Similarly below)
One reason to not care is that we don't particularly want to have to adjust this test if the error message changes.

Reply:
Change request addressed